packages/api/endpoints/rules.js

```diff
@@ -120,11 +124,18 @@ async function put({ params: { name }, body }, res) {
     // Remove all fields from the existing rule that are not supplied in body
     // since body is expected to be a replacement rule, not a partial rule
     const fieldsToDelete = Object.keys(oldRule).filter((key) => !(key in body));
-    const newRule = await model.update(oldRule, body, fieldsToDelete);
-
-    if (inTestMode()) await addToLocalES(newRule, indexRule);
 
-    return res.send(newRule);
+    return await dbClient.transaction(async (trx) => {
+      const newRule = await model.update(oldRule, body, fieldsToDelete);
+      const newPostgresRecord = await translateApiRuleToPostgresRule(newRule, dbClient);
+
+      if (inTestMode()) await addToLocalES(newRule, indexRule);
+      await trx(tableNames.rules)
+        .insert(newPostgresRecord)
+        .onConflict('name')
+        .merge();
+      return res.send(newRule);
```

Comment:
Again, we need to do the Postgres write before the Dynamo write. And let's move the ES write and `return` outside of the transaction